pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -315,22 +318,28 @@ func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Co
 		return ctrl.Result{}, conversionErr
 	}
 
-	newMapiMachine.SetOwnerReferences(newMAPIOwnerReferences)
-
 	for _, warning := range warns {
 		logger.Info("Warning during conversion", "warning", warning)
 		r.Recorder.Event(mapiMachine, corev1.EventTypeWarning, "ConversionWarning", warning)
 	}
 
+	newMapiMachine.SetNamespace(r.MAPINamespace)
+	newMapiMachine.SetOwnerReferences(newMAPIOwnerReferences)
+
 	if mapiMachine != nil {
 		newMapiMachine.SetResourceVersion(util.GetResourceVersion(mapiMachine))
-		newMapiMachine.Spec.Labels = util.MergeMaps(mapiMachine.Spec.Labels, newMapiMachine.Spec.Labels)
-		newMapiMachine.Labels = util.MergeMaps(mapiMachine.Labels, newMapiMachine.Labels)
+		// newMapiMachine.Spec.Labels = util.MergeMaps(mapiMachine.Spec.Labels, newMapiMachine.Spec.Labels)
```

Comment:
Do we want a TODO to work this out later? Why have we stopped this with labels? :)

Reply:
This wasn't required to achieve the updating the mirror machine, as we do this copying in the conversion. I'll remove the comment.